zipp.py

```diff
@@ -195,16 +193,18 @@ def joinpath(self, add):
 
     @staticmethod
     def _implied_dirs(names):
-        return more_itertools.unique_everseen(
-            parent + "/"
-            for name in names
-            for parent in _parents(name)
-            if parent + "/" not in names
-        )
+        seen = set(names)
+        for parent in (
+            p + posixpath.sep
+            for p in itertools.chain.from_iterable(map(_parents, names))
+            if p + posixpath.sep not in seen
+        ):
+            seen.add(parent)
+            yield parent
 
     @classmethod
     def _add_implied_dirs(cls, names):
-        return names + list(cls._implied_dirs(names))
+        return itertools.chain(names, cls._implied_dirs(names))
```

Comment:
FYI: Because this change iterates over names twice, it would break if `namelist()` ever started returning an iterable. My intention was to honor it as if it were an iterable (and not require it to be a sequence).

Reply:
This was causing a test failure as well so I've reverted it.